<?xml version="1.0"?>

<!-- Coordinates are: x forward, y left, z up; rotation is: roll right, pitch forward, yaw left -->
<!-- See http://www.ros.org/reps/rep-0103.html for details -->

<robot name="stereo_test_rig" xmlns:xacro="http://ros.org/wiki/xacro">

  <!-- General constants -->
  <xacro:property name="PI" value="3.1415926535897931"/>

  <!-- Materials -->
  <material name="gray"><color rgba="0.5 0.5 0.5 1.0"/></material>
  <material name="green"><color rgba="0.5 1.0 0.5 1.0"/></material>

  <!-- Base link -->
  <link name="base_link">
    <visual>
      <origin xyz="0 -0.06 0"/>
      <geometry><box size="0.08 0.15 0.005"/></geometry>
      <material name="gray"/> <!-- board -->
    </visual>
    <visual>
      <origin xyz="-0.03 0 -0.02"/>
      <geometry><box size="0.02 0.02 0.03"/></geometry>
      <material name="green"/> <!-- left camera -->
    </visual>
    <visual>
      <origin xyz="0.03 0 -0.02"/>
      <geometry><box size="0.02 0.02 0.03"/></geometry>
      <material name="gray"/> <!-- right camera -->
    </visual>
  </link>

  <!-- Left camera frame -->
  <link name="left_camera_frame" />
  <joint name="base_to_left_camera" type="fixed">
    <origin xyz="-0.03 0 0" rpy="${PI} 0 0"/>
    <parent link="base_link"/>
    <child link="left_camera_frame"/>
  </joint>

  <!-- Right camera frame -->
  <link name="right_camera_frame" />
  <joint name="base_to_right_camera" type="fixed">
    <origin xyz="0.03 0 0" rpy="${PI} 0 0"/>
    <parent link="base_link"/>
    <child link="right_camera_frame"/>
  </joint>

</robot>
